libred/red.c

```diff
@@ -459,6 +477,132 @@ static int call_posix_spawnp(pid_t *restrict pid, const char *restrict file,
 }
 #endif
 
+
+#define redirect_hardcoded(path, replacement)                         \
+  patterns[0] = "^/usr/bin/" path "$";                                \
+  patterns[1] = "^/bin/" path "$";                                    \
+  patterns[2] = "^/usr/sbin/" path "$";                               \
+  for(int i = 0; i < 3; ++i){                                         \
+    compiled = pcre_compile(patterns[i], 0, &err_ptr,                 \
+                            &err_offset, NULL);                       \
+                                                                      \
+    if (!compiled) {                                                  \
+      fprintf(stderr, "Could not compile '%s'\n", path);              \
+      exit(1);                                                        \
+    }                                                                 \
+                                                                      \
+    match = pcre_exec(compiled, NULL, original, strlen(original), 0,  \
+                          0, vector, 3);                              \
+                                                                      \
+    if (match >= 0) {                                                 \
+      int fd = mkstemp(error_template);                               \
+      if (fd == -1) {                                                 \
+        perror("tuscan: mkstemp");                                    \
+        exit(1);                                                      \
+      }                                                               \
+      dprintf(fd, "hardcoded path\n%ld\n%s\n",                        \
+          (long)getpid(), path);                                      \
+      if (close(fd) == -1) {                                          \
+        perror("tuscan: close");                                      \
+        exit(1);                                                      \
+      }                                                               \
+      return xstr(replacement);                                       \
+    }                                                                 \
+  }
+
+
+static const char *transform_path(const char *original){
+  const char *err_ptr;
+  int err_offset;
+  pcre *compiled;
+  int vector[3] = {0};
+  int match;
+  char *patterns[4];
+  for(int i = 0; i < 4; i++){
```

Comment:
Either use `sizeof(patterns) / sizeof(patterns[0])` for the size of array or define this as macro.